File: spec/python/spec/test_eof_exception_bits_le2.py
import unittest
import kaitaistruct

from testformats.eof_exception_bits_le2 import EofExceptionBitsLe2

class TestEofExceptionBitsLe2(unittest.TestCase):
    def test_eof_exception_bits_le2(self):
        with EofExceptionBitsLe2.from_file('src/nav_parent_switch.bin') as r:
            with self.assertRaisesRegexp(EOFError, u"^requested 3 bytes, but only 2 bytes available$"):
                r._read()

            self.assertEqual(r._io.pos(), 1)

            self.assertEqual(r.pre_bits, 0x01)
            self.assertEqual(r._debug['fail_bits']['start'], 1)
            self.assertFalse(hasattr(r, 'fail_bits'))
            self.assertFalse(hasattr(r._debug['fail_bits'], 'end'))

File: spec/java/src/io/kaitai/struct/spec/TestEofExceptionBitsLe2.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.EofExceptionBitsLe2;
import org.testng.annotations.Test;
import static org.testng.Assert.*;

import org.testng.Assert.ThrowingRunnable;
public class TestEofExceptionBitsLe2 extends CommonSpec {
    @Test
    public void testEofExceptionBitsLe2() throws Exception {
        EofExceptionBitsLe2 r = EofExceptionBitsLe2.fromFile(SRC_DIR + "nav_parent_switch.bin");
        assertThrowsEofError(new ThrowingRunnable() {
            @Override
            public void run() throws Throwable {
                r._read();
            }
        });
        assertIntEquals(r._io().pos(), 1);
        assertIntEquals(r.preBits(), 1);
        assertTrue(r._attrStart.containsKey("failBits"));
        assertIntEquals(r._attrStart.get("failBits"), 1);
        assertIntEquals(r.failBits(), 0); // actually uninitialized, hence 0 in Java
        assertFalse(r._attrEnd.containsKey("failBits"));
    }
}
